test/arcade/patch/agent/cell/PatchCellTest.java

```diff
@@ -105,4 +117,128 @@ public void addCycle_givenCycles_appendValues() {
         assertEquals(5, cell.getCycles().get(2));
         assertEquals(3, cell.getCycles().size());
     }
+
+    @Test
+    public void step_calledWhenAgeGreaterThanApoptosisAge_setsApoptoticState() {
+        PatchModule module = mock(PatchModule.class);
+        doReturn(0.0).when(parametersMock).getDouble(any(String.class));
+        doReturn(0).when(parametersMock).getInt(any(String.class));
+        doReturn(10.0).when(parametersMock).getDouble("APOPTOSIS_AGE");
+        int age = 11;
+        ArrayList<State> relevantStates = new ArrayList<>();
+        relevantStates.add(State.QUIESCENT);
+        relevantStates.add(State.MIGRATORY);
+        relevantStates.add(State.PROLIFERATIVE);
+
+        for (State state : relevantStates) {
+            PatchCellContainer container =
+                    new PatchCellContainer(
+                            cellID,
+                            cellParent,
+                            cellPop,
+                            age,
+                            cellDivisions,
+                            state,
+                            cellVolume,
+                            cellHeight,
+                            cellCriticalVolume,
+                            cellCriticalHeight);
+            PatchCell cell = spy(new PatchCellMock(container, locationMock, parametersMock));
+            cell.processes.put(Domain.METABOLISM, metabolismMock);
+            cell.processes.put(Domain.SIGNALING, signalingMock);
+            cell.module = module;
+            doAnswer(
+                            invocationOnMock -> {
+                                cell.state = invocationOnMock.getArgument(0);
+                                return null;
+                            })
+                    .when(cell)
+                    .setState(any(State.class));
+
+            cell.step(simMock);
+
+            assertEquals(State.APOPTOTIC, cell.getState());
+        }
+    }
+
+    @Test
+    public void step_calledWhenAgeLessThanApoptosisAge_doesNotSetState() {
+        PatchModule module = mock(PatchModule.class);
+        doReturn(0.0).when(parametersMock).getDouble(any(String.class));
+        doReturn(0).when(parametersMock).getInt(any(String.class));
+        doReturn(10.0).when(parametersMock).getDouble("APOPTOSIS_AGE");
+        int age = 9;
+        ArrayList<State> relevantStates = new ArrayList<>();
```

Comment:
What is considered a relevant state vs. not? I _think_ any state other than `APOPTOTIC` would trigger this, in which case relevant states would be the full set of states minus `APOPTOTIC`?

Reply:
The distinction here is that `necrotic`, `apoptotic`, and `senescent` are by design terminal states - which is more relevant to #33. I think that `necrotic` and `senescent` behaviors still need to be implemented though...